scripts/kconfig/hardenconfig.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2019 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+from kconfiglib import Symbol, standard_kconfig, standard_config_filename
+
+import csv
+import os
+
+
+def hardenconfig(kconf):
+    kconf.load_config()
+
+    hardened_kconf_filename = os.environ['ZEPHYR_BASE'] + os.sep +  \
+        "scripts" + os.sep + "kconfig" + os.sep + "hardened.csv"
+
+    options = compare_with_hardened_conf(kconf, hardened_kconf_filename, {})
+
+    display_results(options)
+
+
+class Option(object):
+
+    def __init__(self, name: str, recommended: str, current=None, symbol=None):
+        self.name = name
+        self.recommended = recommended
+        self.current = current
+        self.symbol = symbol
+
+        if current is 'NA':
+            self.result = 'NA'
+        elif recommended == current:
+            self.result = 'PASS'
+        else:
+            self.result = 'FAIL'
+
+
+def compare_with_hardened_conf(kconf, hardened_kconf_filename,  options):
+    options = {}
```

Comment:
This change will not be seen outside the function btw. Variables in Python are references, like pointers in C.

Reply:
> > @ulfalizer thanks for reviewing. I'll apply changes on top of the first commit, I'm trying to preserve the initial work from Lauren, is it ok for you ?
> 
> Fine with me. Another option would be a `Co-authored-by:` tag.
> 
> Could squash just the fixups into a single commit otherwise.

Yeah, I was keeping fixes into small commits, but for the sake of readability I'll squash them.